transcom/mymove
# B-21365 Customer UB Shipment Card - INT

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21365)

## Summary

Note: As there is still no way to create a UB shipment strictly with the UI, the work done in this BL item is not feature flagged off. All work here is only seen after a UB shipment is created via the createMTOShipment endpoint with either swagger or postman.

This work is to refine the customer UI for the UB shipment card, shipment container, and shipment list. A bonus of us reusing the HHG components is that some of the functionality just falls in - such as editing/deleting shipments.

### How to test

1. Access the milmove app as a customer, create a move and an HHG shipment. Do not complete the move.
2. Using the createMTOShipment endpoint, create a UB or two.
```
{
  "moveTaskOrderID": "yourMoveIDHere",
  "shipmentType": "UNACCOMPANIED_BAGGAGE",
  "requestedPickupDate": "2024-10-04",
  "pickupAddress": {
      "streetAddress1": "123 Main St",
      "city": "Nicholasville",
      "state": "KY",
      "postalCode": "40356",
      "country": "US",
      "county": "JESSAMINE"
   }
}
```
You should see the shipments appear now:
![image](https://github.com/user-attachments/assets/a4ab9bc4-afc9-4455-8e44-6a2ad76b2df7)
3. Click the edit button for all UBs you created, and fill out all the information as you would with an HHG:
![image](https://github.com/user-attachments/assets/90b52d1b-6a60-461e-9ec6-55aca2385417)

4. Click the Review and submit button:
![image](https://github.com/user-attachments/assets/28c650b1-5f2a-4c16-af72-095105def90c)

5. You should see the normal HHG shipment card:
![image](https://github.com/user-attachments/assets/513e6657-2b58-4daa-94d5-a2d2e3e35f2d)

6. You should also see the new UB shipment card, which reuses the HHG shipment card component so it will look very similar in structure, but be labeled as UB:
![image](https://github.com/user-attachments/assets/c1d20933-65d8-4029-bac0-d941b35d39cd)

7. Extra credit: since we are reusing so much of the HHG functionality for this, test out that deleting the shipment also works. Technically not in scope for this PR but it's nice that it works.

### Frontend

![image](https://github.com/user-attachments/assets/9ed1fb44-dc23-44b7-a0a4-563e095470ff)
![image](https://github.com/user-attachments/assets/2b39a93e-96c6-45b0-9ae6-e73b2e7bf833)
